src/libopensc/pkcs15.c

```diff
@@ -2432,6 +2475,46 @@ sc_pkcs15_read_file(struct sc_pkcs15_card *p15card, const struct sc_path *in_pat
 			}
 			len = head-data;
 		}
+		else if (file->ef_structure == SC_FILE_EF_LINEAR_VARIABLE) {
+
+			u8 offset_buffer[] = {0x54, 0x02, 0x00, 0x00};
+			struct sc_apdu apdu;
+			uint16_t offset_u16 = 0;
+			u8 *data_do;
+			size_t data_do_len;
+
+			do {
+
+				offset_buffer[2] = (u8) (offset_u16 >> 8);
+				offset_buffer[3] = (u8) (offset_u16 & 0x00ff);
+
+				sc_format_apdu(p15card->card, &apdu, SC_APDU_CASE_4_SHORT, 0xB3, in_path->index, 4);
+				
+				apdu.data = offset_buffer;
+				apdu.datalen = 4;
+				apdu.lc = 4;
```

Comment:
maybe `sizeof(offset_buffer)` would be more readable, but this is probably about personal taste